web/src/beta/components/fields/SpacingInput/index.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from "react";
+
+import { styled } from "@reearth/services/theme";
+
+import Property from "..";
+
+type SpacingValues = {
+  top: number;
+  left: number;
+  right: number;
+  bottom: number;
+};
+
+type Props = {
+  name?: string;
+  description?: string;
+  value?: SpacingValues;
+  onChange?: (values: SpacingValues) => void;
+};
+
+const SpacingInput: React.FC<Props> = ({ name, description, value, onChange }) => {
+  const [spacingValues, setSpacingValues] = useState<SpacingValues>(
+    value || { top: 0, left: 0, right: 0, bottom: 0 },
+  );
+
+  const handleInputChange = (position: keyof SpacingValues, newValue: string) => {
+    const updatedValues = { ...spacingValues, [position]: newValue };
+    setSpacingValues(updatedValues);
+    onChange?.(updatedValues);
+  };
+
+  function getSpacingPosition(spacingValue: SpacingValues, position: keyof SpacingValues): number {
+    return spacingValue[position];
+  }
```

Comment:
move this out of the component so it doesn't always have to be re-rendered

Reply:
done